File: Tests/Variables/Vars.cpp
#include <iostream>
using namespace std;

int main() {
    int score = 0;

    // Block scope
    {
        int coins = 5;
        int bonus = 10;

        int total = coins * bonus;
        score += total;

        cout << "Inside block: " << total << endl;
    }

    // Outside block
    cout << "Final score: " << score << endl;
    cout << "Coins outside block: " << coins << endl; // Error: coins not defined

    return 0;
}

File: Tests/Variables/Vars.lua
-- Global variable
Score = 0

-- Block scope simulation
do
    local coins = 5
    local bonus = 10

    local total = coins * bonus
    Score = Score + total

    print("Inside block: " .. total)
end

-- Outside block
print("Final score: " .. Score)
-- print("Coins outside block: " .. coins) -- Would error: coins is local
